### ISSUE TITLE
Please fix the M3U Extended information #EXTINF:{number} to have the correct time data.

### ISSUE THREAD

**Issue by 0rkaM** [justin025/onthespot#44]
### Feature Description

As brought up previously in [#31 note](https://github.com/justin025/onthespot/issues/31#issuecomment-2436779350) and [#31 reply](https://github.com/justin025/onthespot/issues/31#issuecomment-2439672216) the M3U file contains the #EXTINF:{_number_} where _number_ is always -1 for each new line. After reviewing all the information, I do not want anymore that _number_ to be a serialized consecutive incremental number starting at _1_, which one of my devices was able to utilize, but was also based on outdated and undocumented information of M3U Extended v0.4.
But, therefore I would like it to contain the information as int he updated [RFC8216 ](https://datatracker.ietf.org/doc/html/rfc8216)and specifically [section-4.3.2.1](https://datatracker.ietf.org/doc/html/rfc8216#section-4.3.2.1) where _number_ would be the duration of the media expressed in integer conform to version less than 3, to be compatible with the great majority of players.

Here is the extract from that section:

> The EXTINF tag specifies the duration of a Media Segment.  It applies
>    only to the next Media Segment.  This tag is REQUIRED for each Media
>    Segment.  Its format is:
> 
>    #EXTINF:<duration>,[<title>]
> 
>    where duration is a decimal-floating-point or decimal-integer number
>    (as described in [Section 4.2](https://datatracker.ietf.org/doc/html/rfc8216#section-4.2)) that specifies the duration of the
>    Media Segment in seconds.  Durations SHOULD be decimal-floating-
>    point, with enough accuracy to avoid perceptible error when segment
>    durations are accumulated.  However, if the compatibility version
>    number is less than 3, durations MUST be integers.  Durations that
>    are reported as integers SHOULD be rounded to the nearest integer.
>    The remainder of the line following the comma is an optional human-
>    readable informative title of the Media Segment expressed as UTF-8
>    text.
>    

the number to be entered in the field can be calculated from the Length metadata / 1000, for ex.:

Metadata Length = 398876
#EXTINF:398, name_of_song

### Justification

This would be in line with what most players can read and enough up to date to not be obsolete any time soon.
And of course, more correct and informative than: -1

**Comment by justin025** [comment]
for 398876 does the spec call for decimal places and if so how many, i.e. 398.876 or 398?

**Comment by 0rkaM** [comment]
_TLDR:_

(a) if most compatibility than: round(398876/1000)=399

(b) if most cutting edge than: (398876/1000)=398.876

I vote for (a)

_Long Explanation:_

Option 1. In your example I would use '399' integer if you want to keep it simple and you consider a priority the compatibility with most player (software or hardware), since as of today most use and all of them can read version lower than 3.

Option 2. If we want to follow the most up-to-date specification, but also lose compatibility with many players out there, then any decimal floating-point number would be correct depending on the length of the playlist: 1 decimal would be good for less than 10 songs, 2 decimals for less than 100 songs, 3 decimals for less than 1000 songs

As the [RFC8216](https://datatracker.ietf.org/doc/html/rfc8216) specification says:

>the duration can be expressed as 'a decimal-floating-point or decimal-integer number
 that specifies the duration of the
>    Media Segment in seconds.
> 

Also:

> Durations SHOULD be decimal floating-point, with enough accuracy to avoid perceptible error when segment
 durations are accumulated. However, if the compatibility version number is less than 3, durations MUST be integers. Durations that are reported as integers SHOULD be rounded to the nearest integer.

So I would use either 399 for integer or 398.876 for floating-point and nothing else in between, because even if many playlist are fifty, or less songs, there are still a good quantity of them that are in the hundred plus, so to account for those playlist and stay within spec ('enough accuracy to avoid perceptible error'), so ’round half to even’ for integer and in the case of floating-point I would use three decimal digit, which make it also simple by dividing by 1000.

Ultimately it come down to what is key: compatibility of latest (cutting-edge) specification.

P.S.: Python rounds to the nearest even number when the number to be rounded is exactly halfway between two others. This is known as ’round half to even’ or ‘bankers’ rounding’. It’s a strategy used to prevent bias in rounding.

**Comment by justin025** [comment]
Pushed the rounded length in RC-8, if it works and you don't find any other m3u problems feel free to close the issue
